l10n.js

```diff
@@ -219,40 +223,70 @@ document.webL10n = (function(window, document, undefined) {
               parsedRawLinesCallback();
               return;
             }
-            var line = entries.shift();
```

Comment:
Just to address your second bullet point, before the EOL check, we first return ```''``` if reComment is matched so the EOL check should never fire for a comment with the EOL delimiter.

From my reading it also seems that multi-line comments in the way you have described are not part of the properties file spec?